### ISSUE TITLE
Volar not recognizing components globally registered (in main.ts)

### ISSUE THREAD

**Issue by richardtallent-erm** [johnsoncodehk/volar#2209]
Volar is not finding components if they are registered using `app.component(name, component)` in `main.ts` rather than locally imported. So, right-click for definition, etc. does not work, and the markup for them is highlighted in red instead of green.

I can't find an issue for this in the open or closed issues. Is there a workaround for this?

**Comment by johnsoncodehk** [comment]
See https://github.com/johnsoncodehk/volar/issues/1897#issuecomment-1257149754

**Comment by richardtallent-erm** [comment]
I'm using Volar in Takeover mode. The components themselves are normal `.vue` SFC files, but they are globally registered in `main.ts`.

I found the correct answer at the link below, which involves creating a whole *new* file where I have to re-import all shared Vue components and export a `GlobalComponents` type listing them:

https://github.com/johnsoncodehk/volar/blame/1b90234ec6f10a3c0f080d0310711c2cd8de02dd/extensions/vscode-vue-language-features/README.md#L95-L100

I was able to make that work, but it's a chore to maintain. And the solution is sub-optimal, as doing "Go to definition" / F12 doesn't bring me directly to the `.vue` file, it brings up an embedded dialog listing both the `.vue` file and the corresponding `components.d.ts` line, but the latter is selected, not the former, so it takes another two clicks to open the SFC (once to open the accordion for the `.vue` file, then another to open the file itself).
